### PR TITLE
Consolidate Console messages and Deprecation guidelines 

### PR DESCRIPTION
**Pull request by alex-page**
I don't think this page is high value. Linting should do the essentials here.

### CODE REVIEW

**Review by sam-b-rose — APPROVED** [review]
🧹 🗒️ ✨

**Review by kyledurand — APPROVED** [review]

**Review by chloerice — COMMENTED** [review]
> Linting should do the essentials here.

These guidelines help folks on our team and other contributors write deprecation notices for our package consumers that meet our content standards when a prop or component is going to be removed in a future release. (they're not part of the style guide, they're repo only guidelines). I'm not familiar with our project-specific linting rules. Do we have a way of adding deprecated props or components to our linting rules?

**Comment by alex-page** [comment]
@chloerice good point. What do you think about renaming and focusing this on deprecation and remove all console log guidance.

File: documentation/Deprecation guidelines.md [hunk 1]
```diff
@@ -5,12 +5,21 @@
...
 - Use the `@deprecated` doc tag for props and components ([for example, add above where the component is defined](https://github.com/Shopify/polaris/blob/8e49e4c65fbbf25d40617ba2d0ff0b3747320f17/src/components/Navigation/components/UserMenu/UserMenu.tsx#L27)), state the reason, and upgrade path
-- Add a console warning for any respective `@deprecated` doc tag, state the reason, and state the upgrade path
-  - Add the console warning in the constructor for classes and in the render method for functional components
-  - [Guidelines for writing good console messages](https://github.com/Shopify/polaris/blob/main/documentation/Console%20messages.md)
 - For significant deprecations, add a section to the component documentation with rationale. State the upgrade path and include a link to a new component if applicable
 - Call out deprecations in our [changelog](https://github.com/Shopify/polaris/blob/main/documentation/Versioning%20and%20changelog.md)
 
+## Deprecation console message guidelines
+
+Use `warn` for messages `Deprecation` messages.
```
Comment (chloerice) [inline documentation/Deprecation guidelines.md]:
```suggestion
Use `warn` for `Deprecation` messages.
```

File: documentation/Deprecation guidelines.md [hunk 2]
```diff
@@ -5,12 +5,21 @@
...
+
+```js
+console.warn(
+  'Deprecation: The `title` property on Tabs has been deprecated. Use `content` instead.',
+);
+```
+
+Use `Deprecation` prefix for messages that warn developers when component APIs are in the process of being replaced. It’s important to notify developers of breaking changes ahead of their release.
```
Comment (chloerice) [inline documentation/Deprecation guidelines.md]: Should we move this to the top of the page as a way to introduce the content? It feels redundant down here.

```suggestion
Deprecation messages warn developers when component APIs are in the process of being replaced. It’s important to notify developers of breaking changes ahead of their release.
```

**Review by chloerice — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**